### DISCUSSION TITLE
Object detection in video

### DISCUSSION THREAD

**Discussion by chantzeleong** [ultralytics/yolov5#1662] [General]
How would the code be for object detection in video files? Any preferred video format for better accuracy?

**Comment by glenn-jocher** [comment]
@chantzeleong you might want to read the README:
https://github.com/ultralytics/yolov5#inference

Code is:
```
python detect.py --source video.mp4
```

**Reply by MGIbrahim** [reply]
- After training a custom dataset on images, can I use the trained model to also detect videos? 
- If yes, can I do the video inference by loading the model from PyTorch instead of using the detect.py file.

**Reply by glenn-jocher** [reply]
@MGIbrahim 👋 Hello! Thanks for asking about **handling inference results**. YOLOv5 🚀 [PyTorch Hub](https://github.com/ultralytics/yolov5/issues/36) models allow for simple model loading and inference in a pure python environment without using `detect.py`. 

### Simple Inference Example

This example loads a pretrained YOLOv5s model from PyTorch Hub as `model` and passes an image for inference. `'yolov5s'` is the YOLOv5 'small' model. For details on all available models please see the [README](https://github.com/ultralytics/yolov5#pretrained-checkpoints). Custom models can also be loaded, including custom trained PyTorch models and their [exported](https://github.com/ultralytics/yolov5/issues/251) variants, i.e. ONNX, TensorRT, TensorFlow, OpenVINO YOLOv5 models.

```python
import torch

# Model
model = torch.hub.load('ultralytics/yolov5', 'yolov5s')  # yolov5n - yolov5x6 official model
#                                            'custom', 'path/to/best.pt')  # custom model

# Images
im = 'https://ultralytics.com/images/zidane.jpg'  # or file, Path, URL, PIL, OpenCV, numpy, list

# Inference
results = model(im)

# Results
results.print()  # or .show(), .save(), .crop(), .pandas(), etc.
results.xyxy[0]  # im predictions (tensor)

results.pandas().xyxy[0]  # im predictions (pandas)
#      xmin    ymin    xmax   ymax  confidence  class    name
# 0  749.50   43.50  1148.0  704.5    0.874023      0  person
# 2  114.75  195.75  1095.0  708.0    0.624512      0  person
# 3  986.00  304.00  1028.0  420.0    0.286865     27     tie

results.pandas().xyxy[0].value_counts('name')  # class counts (pandas)
# person    2
# tie       1
```
<img width="550" src="https://user-images.githubusercontent.com/26833433/149679662-b2021c78-59b2-4f52-b768-1e5161537c85.png">

See [YOLOv5 PyTorch Hub Tutorial](https://github.com/ultralytics/yolov5/issues/36) for details.

Good luck 🍀 and let us know if you have any other questions!

**Reply by muhammadumair2019** [reply]
@MGIbrahim how did you do it then? video inference by loading model with pytorch not giving results. What should i do?

**Reply by MGIbrahim** [reply]
@muhammadumair2019 I had to divide the video into frames and pass it into the trained model using pytorch. I then would re-construct the video back from the frames after passing it to model.

**Reply by muhammadumair2019** [reply]
@MGIbrahim Thanks for your reply. But, I have to give a live video to the model for detection. In this case, it is a long process.
Do you know any faster way?

**Reply by MGIbrahim** [reply]
Maybe this could help? https://github.com/ultralytics/yolov5/issues/2045

**Reply by muhammadumair2019** [reply]
Thankyou brother. It is very helpful.
